src/chainstate/stacks/db/blocks.rs

```diff
@@ -5272,28 +5273,45 @@ impl StacksChainState {
             _ => false, // unused
         };
 
-        let current_tip =
-            StacksChainState::get_parent_index_block(current_consensus_hash, current_block);
-        let res = match self.with_read_only_clarity_tx(&NULL_BURN_STATE_DB, &current_tip, |conn| {
+        let current_tip = match mempool_admission_check {
+            TransactionAnchorMode::OnChainOnly | TransactionAnchorMode::Any => {
+                StacksChainState::get_parent_index_block(current_consensus_hash, current_block)
+            }
+            TransactionAnchorMode::OffChainOnly => {
+                if let Some(unconfirmed_state) = &self.unconfirmed_state {
+                    if unconfirmed_state.is_readable() {
+                        unconfirmed_state.unconfirmed_chain_tip
+                    } else {
+                        StacksChainState::get_parent_index_block(
+                            current_consensus_hash,
+                            current_block,
+                        )
+                    }
+                } else {
+                    StacksChainState::get_parent_index_block(current_consensus_hash, current_block)
```

Comment:
We want to avoid using `unwrap` in non-test code, which is why I wrote it like this

Reply:
but, that's why i suggested to first check `self.unconfirmed_state.is_some()`... it should be fine to unwrap after checking that, no? I think what we don't want is unchecked unwraps.

Reply:
Please don't `.unwrap()` in network-reachable code.

Reply:
Even if you check it, you still shouldn't unwrap? Is that because there is some way for the unwrap to fail after checking `is_some`? Or because it is an error prone construction and we might one day lose the `is_some` check?

Reply:
The latter -- it's an error-prone construction.